sources/pmwebapi_test.go

```diff
@@ -55,49 +55,29 @@ func TestUnmarshal(t *testing.T) {
 	}
 }
 
-func TestLabelTypes(t *testing.T) {
-	units := ""
-	pcpType := ""
-	name := "kal"
-	response := typeLabel(units, pcpType, name)
-	expected := "kal"
-	if response != expected {
-		t.Fatalf("Failed when units and type strings were empty. Expected: %s, Got: %s", expected, response)
-	}
-
-	units = ""
-	pcpType = ""
-	name = ""
-	response = typeLabel(units, pcpType, name)
-	expected = ""
-	if response != expected {
-		t.Fatalf("Failed when all strings were empty. Expected: %s, Got: %s", expected, response)
-	}
+func TestWeirdMetrics(t *testing.T) {
 
-	units = "kal"
-	pcpType = "CoUnTeR"
-	name = "kal"
-	response = typeLabel(units, pcpType, name)
-	expected = "kal_kal_total"
-	if response != expected {
-		t.Fatalf("Failed when type was case sensitive. Expected: %s, Got: %s", expected, response)
-	}
-
-	units = "tery"
-	pcpType = "gagage"
-	name = "kal"
-	response = typeLabel(units, pcpType, name)
-	expected = "kal_tery"
-	if response != expected {
-		t.Fatalf("Failed when type is any other than counter or empty. Expected: %s, Got: %s", expected, response)
+	name := []string{"count / sec", "Kbyte", "nanosec", "millisec", "failcnt", "byte", "megabytess", "Mbyte", "sec", "bytes_byte", "kilobytess"}
+	expected := []string{"count_per_second", "kilobytes", "nanoseconds", "milliseconds", "failcount", "bytes", "megabytes", "megabytes", "seconds", "bytes", "kilobytes"}
+	var n = 0
```

Comment:
Remember, this isn't C... High-level languages have come a long way since C and we can now use handy tools like `range`. Go makes this even easier because you can get both the iteration counter and the array item in a single line/call.